src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js

```diff
@@ -37,10 +38,31 @@ export default ajaxCaller(class LaunchAnalysisModal extends Component {
     }
   }
 
+  getEntities() {
+    const { entitySelectionModel: { type, selectedEntities }, processSingle, config: { rootEntityType } } = this.props
+
+    if (!processSingle) {
+      return Utils.switchCase(type,
+        [EntitySelectionType.chooseRows, () => _.keys(selectedEntities)],
+        [EntitySelectionType.processMergedSet, () => _.flow(
+          _.flatMap(`attributes.${rootEntityType}s.items`),
+          _.map('entityName')
+        )(selectedEntities)],
+        [Utils.DEFAULT, () => selectedEntities]
+      )
+    }
+  }
+
   render() {
-    const { onDismiss, entitySelectionModel: { type, selectedEntities } } = this.props
+    const { onDismiss, entitySelectionModel: { type }, entityMetadata, config: { rootEntityType }, processSingle } = this.props
     const { launching, message, multiLaunchCompletions, launchError, multiLaunchErrors } = this.state
-    const entityCount = _.size(selectedEntities)
+    const entities = this.getEntities()
+    const entityCount = Utils.cond(
+      [processSingle, () => 1],
+      [type === EntitySelectionType.processAll, () => entityMetadata[rootEntityType].count],
+      [_.isArray(entities), () => _.uniq(entities).length],
```

Comment:
In which case is `entities` an array? I'm wondering if this is always doing the right thing. If the root entity type is a singular type and the selected entity is a set, then the effective number of entities (and therefore the number of workflows) will be the number of entities in the set.

Reply:
Talked about this with faces last week. Gonna merge for now, this could use a refactor eventually.